tests/fips/openssl/openssl_pubkey_dsa.pm

```diff
@@ -50,11 +56,21 @@ sub run {
         # A source of random numbers is required for DSA, so get message digest first
         assert_script_run "openssl dgst -$dgst_alg $file_raw | awk '{print $2}' > $file_dgt";
 
-        # Sign message digest with private key
-        assert_script_run "openssl dgst -dss1 -sign $dsa_prikey $file_dgt > $file_sig";
+        # Sign message digest with private key and verify signature with public key
+        if ($openssl_version eq "1.1") {
 
-        # Verify signature with public key
+        # openssl version >= 1.1.x
+        # Sign message digest with private key and verify signature with public key (sha256)
+        assert_script_run "openssl dgst -sha256 -sign $dsa_prikey $file_dgt > $file_sig";
+        validate_script_output "openssl dgst -sha256 -verify $dsa_pubkey -signature $file_sig $file_dgt", sub { m/Verified OK/ };
+        }
+
+        else {
+        # openssl version <= 1.0.x
+        # Sign message digest with private key and verify signature with public key (dss1)
+        assert_script_run "openssl dgst -dss1 -sign $dsa_prikey $file_dgt > $file_sig";
         validate_script_output "openssl dgst -dss1 -verify $dsa_pubkey -signature $file_sig $file_dgt", sub { m/Verified OK/ };
```

Comment:
If I was you, I would like to solve this issue with a single "assert_scritp_run", to avoid redundancy:
```
my $algo = ($openssl_version eq "1.1") ? "sha256" : "dss1";
assert_script_run "openssl dgst -$algo -sign $dsa_prikey $file_dgt > $file_sig";
validate_script_output "openssl dgst -$algo -verify $dsa_pubkey -signature $file_sig $file_dgt", sub { m/Verified OK/ };
```

Reply:
@wnereiz  It is a nice suggestion to refine and avoid redundancy. I will try to modify it again. Thanks.